apiserver/facades/agent/upgradesteps/upgradesteps.go

```diff
@@ -114,3 +182,23 @@ func (api *UpgradeStepsAPI) getMachine(canAccess common.AuthFunc, tag names.Mach
 	}
 	return machine, nil
 }
+
+func (api *UpgradeStepsAPI) getUnit(canAccess common.AuthFunc, tag names.UnitTag) (Unit, error) {
+	if !canAccess(tag) {
+		logger.Criticalf("getUnit kind=%q, name=%q", tag.Kind(), tag.Id())
+		return nil, common.ErrPerm
+	}
+	entity, err := api.st.FindEntity(tag)
+	if err != nil {
+		logger.Criticalf("unable to find entity %q", tag, err)
+		return nil, err
+	}
+	// The authorization function guarantees that the tag represents a
+	// unit.
+	var unit Unit
+	var ok bool
+	if unit, ok = entity.(Unit); !ok {
+		return nil, errors.NotValidf("unit entity")
+	}
+	return unit, nil
```

Comment:
Maybe a bit cleaner:

```suggestion
	if unit, ok = entity.(Unit); ok {
	        return unit, nil
	}
	return nil, errors.NotValidf("unit entity")
```